pkg/nfd-master/nfd-master.go

```diff
@@ -408,6 +411,10 @@ func filterFeatureLabels(labels Labels, extraLabelNs map[string]struct{}, labelW
 				klog.Errorf("Namespace %q is not allowed. Ignoring label %q\n", ns, label)
 				continue
 			}
+			if isNamespaceDenied(ns, wildcardDeniedLabelNs, normalDeniedLabelsNs) {
+				klog.Errorf("Namespace %q is not allowed. Ignoring label %q\n", ns, label)
+				continue
+			}
```

Comment:
I think this logic is now broken because we don't change the handling of extraLabelNs above. We basically still deny everything. The logic should be so that extraLabelNs can be used to override DeniedLabelNs. I.e. label is denied if namespace is in DeniedLabelNs, unless specifically allowed with extraLabelNs. Makes sense?

Reply:
Yes makes sense, good point. I was actually thinking of the old logic. But when I reviewed our conversation, I see that the logic is broken. I'll update it